File: 2021/day_01/part_1.rb
# frozen_string_literal: true

require_relative '../advent'

report = []
increased_count = 0

input('day_01').each_with_index do |line, index|
  value = line.to_i
  report[index] = value

  next if index.zero?

  increased_count += 1 if value > report[index - 1]
end

puts "Increased #{increased_count} times"

# Input 1: 7
# Input 2: 1602

File: 2021/day_01/part_1.rs
use std::fs;

// Input 1: 7
// Input 2: 1602

fn main() {
    let mut report = Vec::new();

    let contents = fs::read_to_string("input_2.txt")
        .expect("Something went wrong reading the file");

    let mut index = 0;
    let mut increased_count = 0;

    for line in contents.trim().split("\n") {
        let value: i32 = line.parse().expect("Expected a number for line");

        report.push(value);

        if index > 0 && value > report[index - 1] {
            increased_count += 1;
        }

        index += 1;
    }

    println!("Increased {} times", increased_count);
}
